src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -397,6 +397,40 @@ ShouldSyncTableMetadata(Oid relationId)
 }
 
 
+/*
+ * ShouldSyncTableMetadataViaCatalog checks if the metadata of a distributed table should
+ * be propagated to metadata workers, i.e. the table is an MX table or reference table.
+ * Tables with streaming replication model (which means RF=1) and hash distribution are
+ * considered as MX tables while tables with none distribution are reference tables.
+ *
+ * ShouldSyncTableMetadataViaCatalog does not use the CitusTableCache, and use only the
+ * catalog tables.
+ */
+bool
+ShouldSyncTableMetadataViaCatalog(Oid relationId)
+{
+	if (!OidIsValid(relationId) || !IsCitusTableViaCatalog(relationId))
+	{
+		return false;
+	}
+
+	char replicationModel = ReplicationModelViaCatalog(relationId);
+	char partitionMethod = PartitionMethodViaCatalog(relationId);
+	bool streamingReplicated = replicationModel == REPLICATION_MODEL_STREAMING;
+	bool mxTable = streamingReplicated && partitionMethod == DISTRIBUTE_BY_HASH;
+	bool citusTableWithNoDistKey = partitionMethod == DISTRIBUTE_BY_NONE;
+
+	if (mxTable || citusTableWithNoDistKey)
```

Comment:
can we move this check to a function and use it in both `ShouldSyncTableMetadataViaCatalog` and `ShouldSyncTableMetadata`?

Reply:
I created the function `ShouldSyncTableMetadataInternal` and used it in both `ShouldSyncTableMetadata` and `ShouldSyncTableMetadataViaCatalog`.

The two parameters `mxTable` and `citusTableWithNoDistKey` are acquired from cache or catalog in these functions, but the decision mechanism is common in the internal helper function.

Note:  GitHub moved @onderkalaci 's comment that used to be in `ShouldSyncTableMetadataViaCatalog` and not in `ShouldSyncTableMetadataInternal`